### PR TITLE
Make the run task honor tests.es properties

### LINKED ISSUE #47797: RunTask should pass settings from system properties starting with test.es.
As we replaced the RunTask with one that works with testclusters we lost the possibility of adding setting from the command line.  

While `-Drun.license=trial` can still be used to change the license, one has no way for instance to run off security.

### PR DESCRIPTION
**Pull request by alpar-t**
Closes #47797

With this PR we now again honor tests.es and tests.heap.size

### CODE REVIEW

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java [hunk 1]
```diff
@@ -36,12 +39,20 @@ public void beforeStart() {
         int debugPort = 8000;
         int httpPort = 9200;
         int transportPort = 9300;
+        Map<String, String> additionalSettings = System.getProperties().entrySet().stream()
+            .filter(entry -> entry.getKey().toString().startsWith(CUSTOM_SETTINGS_PREFIX))
+            .collect(Collectors.toMap(
+                entry -> entry.getKey().toString().substring(CUSTOM_SETTINGS_PREFIX.length()),
+                entry -> entry.getValue().toString()
+            ));
+        System.err.println(additionalSettings.toString());
```
Comment (dimitris-athanasiou) [inline buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java]: Was this intended?

Comment (atorok) [reply buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java]: No it was not, I'm going to remove it

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

**Review by rjernst — APPROVED** [review]
LGTM

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java [hunk 2]
```diff
@@ -36,12 +39,19 @@ public void beforeStart() {
         int debugPort = 8000;
         int httpPort = 9200;
         int transportPort = 9300;
+        Map<String, String> additionalSettings = System.getProperties().entrySet().stream()
```
Comment (mark-vieira) [inline buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java]: We need to track this stuff as an input for incremental build and caching to work as expected. I'd move all this into a `getAdditionalSettings()` method annotated with `@Input`.

**Review by mark-vieira — APPROVED** [review]
LGTM 👍

**Pull request merged**